src/modules/haxe.rs

```diff
@@ -0,0 +1,252 @@
+use super::{Context, Module, ModuleConfig};
+
+use crate::configs::haxe::HaxeConfig;
+use crate::formatter::StringFormatter;
+use crate::formatter::VersionFormatter;
+use serde_json as json;
+
+use regex::Regex;
+const HAXE_VERSION_PATTERN: &str =
+    "(?P<version>(?:\\d+\\.\\d+\\.\\d+(?:-(?:alpha|beta|pre|rc)[\\d\\.+]+)?)|[[:xdigit:]]+)";
+
+/// Creates a module with the current Haxe version
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("haxe");
+    let config = HaxeConfig::try_load(module.config);
+
+    let is_haxe_project = context
+        .try_begin_scan()?
+        .set_files(&config.detect_files)
+        .set_extensions(&config.detect_extensions)
+        .set_folders(&config.detect_folders)
+        .is_match();
+
+    if !is_haxe_project {
+        return None;
+    }
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_meta(|var, _| match var {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "version" => {
+                    let haxe_version = get_haxe_version(context)?;
+                    VersionFormatter::format_module_version(
+                        module.get_name(),
+                        &haxe_version,
+                        config.version_format,
+                    )
+                    .map(Ok)
+                }
+                _ => None,
+            })
+            .parse(None, Some(context))
+    });
+
+    module.set_segments(match parsed {
+        Ok(segments) => segments,
+        Err(error) => {
+            log::warn!("Error in module `haxe`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+fn get_haxe_version(context: &Context) -> Option<String> {
+    let file_contents = context.read_file_from_pwd(".haxerc");
+    if file_contents.is_some() {
+        let raw_json = match file_contents {
+            Some(val) => val,
+            None => panic!("failed to read contents of .haxerc!"),
+        };
```

Comment:
```suggestion
    if let Some(raw_json) = file_contents {
```

Reply:
thanks for suggestion, it's my first dabble into rust territory :)